proj/funcDictionary.py

```diff
@@ -0,0 +1,335 @@
+    plotData.clustersizesAllArray.append(innerData.clustersizes)
+    plotData.diametersAllArray.append(innerData.diameters)
+    plotData.centersAllArray.append(innerData.centers)
+    plotData.eccentritiesAllArray.append(innerData.eccentrities)
+    plotData.degreeMeanAllArray.append(innerData.degreeMean)
+    
+    return plotData
+
+def convert_to_amino_acids(plotData, innerData): # check description
+    """
+    info: convert lengths to the aminoacids
+    input: plotData: PlotData object, that contains the important data and parameters;
+            for more detail see documentation in plotData.py
+        innerData: objects, that store network properties in list # in more detail perhaps
+    output: plotData: object, that stores the network properties, now with updated members
+    """
+
+    xArgsNew = list()
+    clustersizesNew = list()
+    diametersNew = list()
+    eccentritiesNew = list()
+    degreeMeanNew = list()
+
+    for i in range(len(innerData.xArgs)):
+        if i%3 == 0:
+            xArgsNew.append(innerData.xArgs[i])
+            clustersizesNew.append(innerData.clustersizes[i])
+            diametersNew.append(innerData.diameters[i])
+            eccentritiesNew.append(innerData.eccentrities[i])
+            degreeMeanNew.append(innerData.degreeMean[i])
+
+    plotData.clustersizesAllArray.append(clustersizesNew)
+    plotData.diametersAllArray.append(diametersNew)
+    #plotData.centersAllArray.append(centers)
+    plotData.eccentritiesAllArray.append(eccentritiesNew)
+    plotData.degreeMeanAllArray.append(degreeMeanNew)
+    plotData.xArgsAllArray.append(xArgsNew)
+    
+    return plotData # that belongs here, right?
+
+def degree_mean(GSub):
+    """
+    info: calculate the average degree in the graph GSub
+    input: GSub: graph
+    output: deg: average degree number (real number)
+    """
+    deg = 0
+    degs = GSub.degree()
+    degs = list(degs)
+    for el in degs:
+        deg += el[1]/(len(degs))
+    return deg
+
+def makeDegreeDistribution(G): # I think the name does not describe, what that actually does; better change name
+    """
+    info: calculate a list, which contains the sizes (number of nodes) of each connected subgraph (thus each cluster)
+    input: G: graph
+    output: clustersizes: list with the lengths of all clusters in G
+        c: list of degree values of each node
+    """
+
+    # info: make the degree distribution
+    a = G.degree()
+    
+    # info: b: list, where each element is an array: element 0 is the strain and element 1 is the corresponding degree value
+    b = list(a)
+
+    # info: make list c, which only contains the degree values
+    c = list()
+    for i in range(len(b)):
+        # what is this element? ...; Is that later used?
+        c.append(b[i][1])
+        # info: make the clusters, clustersizes
+    clusters = list(nx.connected_components(G))
+    clustersizes = list()
+    for i in range(len(clusters)):
+        clustersizes.append(len(clusters[i]))
```

Comment:
don't use a `for` loop here: `clustersizes = [cluster[i].size for cluster in clusters]` should do the job.